Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py

```diff
@@ -281,15 +288,23 @@ def yield_response() -> Generator:
             mimetype='application/xml'
         )
 
-    def get_url(self):
+    def get_url(self, request_headers: Headers) -> str:
         """
+        Args:
+            request_headers: The request headers
+
         Returns:
             The service URL according to the protocol.
         """
-        if self.http_server:
-            return f'{self.host}:{self.port}'
+        base_url = f'{self.url_scheme}://{self.host}'
+        if request_headers and '/instance/execute' in request_headers.get('X-Request-URI', ''):
+            # if the server rerouting is used, then the X-Request-URI header is added to the request by the server
+            # and we should use the /instance/execute endpoint in the address
+            calling_context = get_calling_context()
+            instance_name = calling_context.get('IntegrationInstance', '')
+            return os.path.join(base_url, 'instance', 'execute', instance_name)
         else:
-            return self.host
+            return f'{base_url}:{self.port}'
```

Comment:
I may be wrong here but I think when not going through the server, we set the protocol to `https`.

Reply:
not sure i follow
if we're not using the server rerouting, then the scheme can be either http or https - depends if a certificate was set in the integration instance
basically it can also be https if they're using reverse proxy to direct the traffic the https traffic to the http integration web app
am i missing something?